quill-core/src/test/scala/io/getquill/context/mirror/MirrorIdiomSpec.scala

```diff
@@ -42,6 +42,28 @@ class MirrorIdiomSpec extends Spec {
     }
   }
 
+  "should not fail with stack overflow" - {
```

Comment:
is this something you observed in a real application?

Reply:
almost. It was only for MirrorIdiom for cases simullar to https://github.com/getquill/quill/issues/747.
I tried to implement batch insert via `infix`. But i fails with stack overflow for lots of inserts

Reply:
you are right. it's just a conner case for MirrorIdiom. I'll reopen pull request for perfomance fixes only

Reply:
You are right. It's just a conner case for MirrorIdiom. I'll reopen it with perfomance fixes only